modules/core/src/main/java/org/apache/ignite/internal/processors/platform/cache/PlatformCache.java

```diff
@@ -1420,9 +1420,10 @@ private Query readTextQuery(BinaryRawReader reader) {
         boolean loc = reader.readBoolean();
         String txt = reader.readString();
         String typ = reader.readString();
+        final int limit = reader.readInt();
         final int pageSize = reader.readInt();
 
-        return new TextQuery(typ, txt).setPageSize(pageSize).setLocal(loc);
+        return new TextQuery(typ, txt, limit).setPageSize(pageSize).setLocal(loc);
```

Comment:
Platforms do not support 'limit' param yet. 
So, attempt to read it can lead an error. 

Let's set a default value here and create a separate ticket for the issue.